app/models/shop.rb

```diff
@@ -24,4 +30,18 @@ def self.last
   def self.all
     DB[:shops].all
   end
+
+  def self.find(id)
+    h = DB[:shops].first(:id => id) #sequel doc uses first for retrieving record
+    return nil if h == nil
+    Shop.new(h) #returns a shop object with hash argument
+  end
+
+  def set(id, name)
+    DB[:shops].where(id: id).update(name: name)
+  end
+
+  def delete(id)
```

Comment:
if you add a method reader for `id` then you don't need to pass `id` as a parameter :)

Reply:
So do I need to pass ID like this? 

attr_accessor :id, :name, :description

Reply:
you can follow the same concept used for the `update` method